pkg/storegateway/series_refs.go

```diff
@@ -720,15 +718,21 @@ func (s *loadingSeriesChunkRefsSetIterator) Next() bool {
 	}
 	nextPostings := s.postingsSetIterator.At()
 
-	var seriesCacheKey indexcache.PostingsKey
+	var cachedSeriesID cachedSeriesForPostingsID
 	if s.skipChunks {
-		// Calculate the cache key before we filter out anything from the postings,
+		var err error
+		// Calculate the cache ID before we filter out anything from the postings,
 		// so that the key doesn't depend on the series hash cache or any other filtering we do on the postings list.
-		// Calculate the postings key only if we'll actually use it.
-		seriesCacheKey = indexcache.CanonicalPostingsKey(nextPostings)
-		if cachedSet, isCached := fetchCachedSeriesForPostings(s.ctx, s.tenantID, s.indexCache, s.blockID, s.matchers, s.shard, seriesCacheKey, s.logger); isCached {
-			s.currentSet = cachedSet
-			return true
+		// Calculate the cache item ID only if we'll actually use it.
+		cachedSeriesID.postingsKey = indexcache.CanonicalPostingsKey(nextPostings)
+		cachedSeriesID.encodedPostings, err = diffVarintSnappyEncode(index.NewListPostings(nextPostings), len(nextPostings))
+		if err != nil {
+			s.logger.Log("msg", "could not encode postings for series cache key", "err", err)
```

Comment:
The level is missing. Please wrap the logger with the Error level (Error because this is not expected to happen, if happens it's a bug).